agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4622,6 +4662,32 @@ bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMess
                    << ", authentication_type: " << std::hex << int(authtype)
                    << ", encryption_type: " << int(enctype) << (fronthaul ? " fronthaul" : "")
                    << (backhaul ? " backhaul" : "") << (teardown ? " teardown" : "");
+
+        if (!fronthaul && !backhaul && !teardown)
+            continue;
```

Comment:
This should be logged as an error. But the `continue` part is good.

In addition, I would also add a check that there's either a single teardown, or one or more front/backhaul M2s. So it should not be allowed that there is no M2 at all, or that a teardown is combined with anything else. Of course, the question is what to do in that situation: tear down? partially configure? I would say to do a teardown. So the check is simple: if teardown is true, then m2_list should have size 1.

Reply:
According new design error cases are ignored.